### ISSUE TITLE
mbo total budget arg

### ISSUE THREAD

**Issue by giuseppec** [mlr-org/mlrMBO#132]
Hi,

we may need a `total.budget` parameter that defines the total number of function evaluations.
Currently the total number of function evaluations depends from `init.design.points`, `iters` AND `interleave.random.points`. Let's assume I want only 100 total function evaluation, then I would configure MBO like:
```r
tot.budget = 100L
n.init = 30L
  ctrl = makeMBOControl(
    init.design.points = n.init,
    iters = tot.budget - n.init
  )
```
However, using e.g. `interleave.random.points = 1` yields in 170 function evaluations because we additionally have another function evaluation for the `interleave.random.points` in each iteration. So we might replace the `iters` argument by a `total.budget` that automatically computes the number of iterations given the `init.design.points` and `interleave.random.points`  (and I think there was another parameter that controls the number of function evaluations) argument?

**Comment by jakob-r** [comment]
It's a valid point. It should be very easy if we accept that we can exceed the tot.budget by a few evluations (for example if the interleaved random points exeed the limit in one iteration).

However it could also be discussed if the budget should be checked before every evaluation which is possible but impractical a bit.

**Comment by berndbischl** [comment]
IMHO we have to write down a formula first, how the budget is computed.
Then go from there.

Let 
i = init design
s = sequential steps
k = proposed points
r = random interleave points

then 
budget = i + s (k + r)

so, is that correct? for every mbo variant?
because if so, the only thing we need to compute is "s".
which is simple. and no, we dont want a few points too many, the budget MUST be a hard barrier, so we need to set s a bit too small.

further questions
a) i am not sure i like the random.interleave in its current definition. but maybe that is another point and does not belong here.
b) we have "final evals". but these should not be counted against the budget as they are not part of the optimization, more of an auto-final-examination of the proposed point.
c) if s was set too small, we might think about how to spend the last evals. or we just igore that for simplicity?

**Comment by tobiaswagner** [comment]
What about defining the random interleave points r not as an integer, but as a portion of k, the number of proposed points. This would allow the remainder of the budget after the floor operation on s to be splitted accordingly. If r is lower than 1, for instance 0.2, it would define that a random point is evaluated every fifth iteration.
________________________________________
Von: Bernd Bischl [notifications@github.com]
Gesendet: Montag, 4. Januar 2016 20:10
An: mlr-org/mlrMBO
Betreff: Re: [mlrMBO] mbo total budget arg (#132)

IMHO we have to write down a formula first, how the budget is computed.
Then go from there.

Let
i = init design
s = sequential steps
k = proposed points
r = random interleave points

then
budget = i + s (k + r)

so, is that correct? for every mbo variant?
because if so, the only thing we need to compute is "s".
which is simple. and no, we dont want a few points too many, the budget MUST be a hard barrier, so we need to set s a bit too small.

further questions
a) i am not sure i like the random.interleave in its current definition. but maybe that is another point and does not belong here.
b) we have "final evals". but these should not be counted against the budget as they are not part of the optimization, more of an auto-final-examination of the proposed point.
c) if s was set too small, we might think about how to spend the last evals. or we just igore that for simplicity?

�

**Comment by berndbischl** [comment]
Lets make a issue for the "random interleave" stuff which is separate from this. Here we should only discuss the budget. So we have to assume here that random interleave works as it is implemented now.

**Comment by jakob-r** [comment]
Will be solved in #137

**Comment by berndbischl** [comment]
done
